public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx

```diff
@@ -0,0 +1,257 @@
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  Output,
+  ViewChild,
+  ViewEncapsulation
+} from '@angular/core';
+import { environment } from 'environments/environment';
+import * as highCharts from 'highcharts';
+import HighchartsReact from 'highcharts-react-official';
+import * as React from 'react';
+import { FunctionComponent } from 'react';
+import * as ReactDOM from 'react-dom';
+import { BaseReactWrapper } from '../../../data-browser/base-react/base-react.wrapper';
+import { baseOptions, GENDER_STRATUM_MAP } from '../../../data-browser/charts/react-base-chart/base-chart.service';
+
+const containerElementName = 'root';
+
+interface State {
+    options: any;
+}
+
+interface Props {
+    genderAnalysis: any;
+    genderCountAnalysis: any;
+    selectedResult: any;
+    domain: string;
+}
+
+export class BioSexChartReactComponent extends React.Component<Props, State> {
+
+  constructor(props) {
+    super(props);
+    this.state = {options: null};
+  }
+
+  componentDidMount() {
+      this.getChartOptions();
+  }
+
+  getChartOptions() {
+    if (this.props.domain === 'fitbit') {
+        this.getFitbitChartOptions();
+    } else {
+        this.getSurveyChartOptions();
+    }
+  }
+
+  getFitbitChartOptions() {
+    baseOptions.chart.type = 'bar';
+    baseOptions.plotOptions.series.pointWidth = 50;
+    baseOptions.yAxis.title.text = 'Participant Count';
+    baseOptions.xAxis.title.text = '';
+    baseOptions.yAxis.title.style.fontSize = '14px';
+    baseOptions.xAxis.title.style.fontSize = '14px';
+    baseOptions.yAxis.title.style.color = '#262262';
+    baseOptions.yAxis.gridLineColor = '#F0F0F0';
+    const {categories, data} = this.prepFitbitCategoriesAndData();
+    baseOptions.xAxis.categories = categories;
+    baseOptions.series[0].data = data;
+    this.setState({options: baseOptions});
+  }
+
+  getSurveyChartOptions() {
+    baseOptions.chart.type = 'column';
+    baseOptions.plotOptions.column.groupPadding = 0.40;
+    baseOptions.plotOptions.series.pointWidth = 50;
+    baseOptions.legend.enabled = true;
+    baseOptions.yAxis.gridLineColor = '#ECF1F4';
+    baseOptions.title.style.color = '#262262';
+    baseOptions.title.style.fontSize = '22px';
+    baseOptions.color = '#2691D0';
+    baseOptions.xAxis.title.text = this.props.genderAnalysis.analysisName;
+    baseOptions.yAxis.title.text = 'Participant Count';
+    this.props.genderAnalysis.results = this.props.genderAnalysis.results.filter(
+              r => r.stratum4 === this.props.selectedResult.stratum4);
+    const {categories, series} = this.prepSurveyCategoriesAndData();
```

Comment:
Use the destructed prop values from the above comment. Also should create a new const instead of modifying props directly. Can store new value in state or might be easier to pass into the function as an arg:
```suggestion
    baseOptions.xAxis.title.text = analysisName;
    baseOptions.yAxis.title.text = 'Participant Count';
    const filteredResults = results.filter(
              r => r.stratum4 === selectedResult.stratum4);
    const {categories, series} = this.prepSurveyCategoriesAndData(filteredResults);
```

Reply:
Changed this.